app/controllers/blogs_controller.rb

```diff
@@ -4,6 +4,7 @@ class BlogsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
 
   before_action :set_blog, only: %i[show edit update destroy]
+  before_action :prohibit_unauthenticated_access, only: %i[edit update destroy]
```

Comment:
```suggestion
  before_action :set_blog, only: %i[show] # 閲覧可能なblogを検索する
  before_action :set_own_blog, only: %i[edit update destroy] # ログインユーザーのblogを検索する
```

↑のようにshowとそれ以外でset_blogを切り分けた方が実装がシンプルになるかもです。

さらに言うと、showのときだけが特殊なので、showだけbefore_actionを使わずに`def show`の中でblogを検索するのも一手です。

```suggestion
  before_action :set_blog, only: %i[edit update destroy] # ログインユーザーのblogを検索する
```